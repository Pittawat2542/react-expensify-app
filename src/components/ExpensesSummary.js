import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";

import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <h1>
    Viewing{" "}
    {expenseCount +
      (expenseCount === 1 || expenseCount === 0
        ? " expense"
        : " expenses")}{" "}
    totalling {numeral(expensesTotal / 100).format("$0,0.00")}
  </h1>
);

const mapStateToProps = state => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(
      selectExpenses(state.expenses, state.filters)
    )
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
