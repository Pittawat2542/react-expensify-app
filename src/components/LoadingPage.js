import React from "react";

export default () => {
  return (
    <div className="loader">
      <img
        className="loader__image"
        src="/images/loader.gif"
        alt="Loading..."
      />
    </div>
  );
};
