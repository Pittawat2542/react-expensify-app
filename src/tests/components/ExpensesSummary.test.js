import { shallow } from "enzyme";
import React from "react";

import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test("should render ExpensesSumamry with 1 expense correctly", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSumamry with multiple expenses correctly", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={243982395} />);
    expect(wrapper).toMatchSnapshot();
});
