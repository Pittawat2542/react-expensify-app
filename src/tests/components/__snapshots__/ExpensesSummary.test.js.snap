// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSumamry with 1 expense correctly 1`] = `
<h1>
  Viewing
   
  1 expense
   
  totalling 
  $1.23
</h1>
`;

exports[`should render ExpensesSumamry with multiple expenses correctly 1`] = `
<h1>
  Viewing
   
  23 expenses
   
  totalling 
  $2,439,823.95
</h1>
`;
